geomstats/geometry/product_riemannian_metric.py

```diff
@@ -174,8 +174,12 @@ def inner_product(
             return gs.sum(gs.stack(inner_prod, axis=1), axis=1)
 
         if point_type == 'matrix':
+            # TODO(ninamiolane): Vectorize this more efficiently
+            tangent_vec_a = gs.to_ndarray(tangent_vec_a, to_ndim=2)
             tangent_vec_a = gs.to_ndarray(tangent_vec_a, to_ndim=3)
+            tangent_vec_b = gs.to_ndarray(tangent_vec_b, to_ndim=2)
             tangent_vec_b = gs.to_ndarray(tangent_vec_b, to_ndim=3)
+            base_point = gs.to_ndarray(base_point, to_ndim=2)
             base_point = gs.to_ndarray(base_point, to_ndim=3)
             inner_products = [metric.inner_product(tangent_vec_a[:, i],
                                                    tangent_vec_b[:, i],
```

Comment:
indentation

Reply:
what about using `tangent_vec_a[..., i, :]` here?

Reply:
done and done